airbyte-integrations/connectors/source-mailgun/source_mailgun/manifest.yaml

```
@@ -1,550 +1,609 @@
+        - id
+      retriever:
+        type: SimpleRetriever
+        requester:
+          $ref: "#/definitions/base_requester"
+          path: /domains
+          http_method: GET
+        record_selector:
+          type: RecordSelector
+          extractor:
+            type: DpathExtractor
+            field_path:
+              - items
+        paginator:
+          type: DefaultPaginator
+          page_token_option:
+            type: RequestOption
+            inject_into: request_parameter
+            field_name: skip
+          pagination_strategy:
+            type: OffsetIncrement
+            page_size: 1000
+      schema_loader:
+        type: InlineSchemaLoader
+        schema:
+          $ref: "#/schemas/domains"
+    events:
+      type: DeclarativeStream
+      name: events
+      primary_key:
+        - id
+      retriever:
+        type: SimpleRetriever
+        requester:
+          $ref: "#/definitions/base_requester"
+          path: /events
+          http_method: GET
+        record_selector:
+          type: RecordSelector
+          extractor:
+            type: DpathExtractor
+            field_path:
+              - items
+        paginator:
+          type: DefaultPaginator
+          page_token_option:
+            type: RequestOption
+            inject_into: request_parameter
+            field_name: from
+          pagination_strategy:
+            type: CursorPagination
+            cursor_value: "{{ response['paging']['next'] if response['items'] }}"
+            stop_condition: "{{ not response.get('items', []) }}"
+      incremental_sync:
+        type: DatetimeBasedCursor
+        cursor_field: timestamp
+        lookback_window: P31D
+        cursor_datetime_formats:
+          - "%s"
+        datetime_format: "%s"
+        start_datetime:
+          type: MinMaxDatetime
+          datetime: >-
+            {{ config.get('start_date', day_delta(-90,
+            format='%Y-%m-%dT%H:%M:%SZ')) }}
+          datetime_format: "%Y-%m-%dT%H:%M:%SZ"
+        end_datetime:
+          type: MinMaxDatetime
+          datetime: "{{ today_utc() }}"
+          datetime_format: "%Y-%m-%d"
+        step: P1M
+        cursor_granularity: PT0.000001S
+      schema_loader:
+        type: InlineSchemaLoader
+        schema:
+          $ref: "#/schemas/events"
+  base_requester:
     type: HttpRequester
-    url_base: |
-      {{
-        'https://api.mailgun.net/v3'
-        if config['domain_region'] == 'US'
-        else 'https://api.eu.mailgun.net/v3'
+    url_base: |-
+      {{        
+        'https://api.eu.mailgun.net/v3'
+        if config.domain_region == 'EU'
+        else 'https://api.mailgun.net/v3'
       }}
-    http_method: "GET"
     authenticator:
-      type: "BasicHttpAuthenticator"
-      username: "api"
+      type: BasicHttpAuthenticator
       password: "{{ config['private_key'] }}"
+      username: "{{ config['username_do_not_edit'] if config['username_do_not_edit'] is defined else 'api'}}"
```

@lmossman would the builder overwrite the authenticator value if we hardcode the username to '`api'`?
```suggestion
      username: "api"
```

---

The authentication component only supports a value of the form `{{ config['key'] }}` when in UI mode, so if this is set to anything else (e.g. `{{ config['username_do_not_edit'] if config['username_do_not_edit'] is defined else 'api'}}` or `api`), then it will load the authenticator component as YAML and won't support switching to the UI unless it is changed to that form

---

thanks for confirming!

@pabloescoder let's go with the simpler approach (hardcode the value to `api`) and accept the connector builder will fallback to YAML for the component.

---

@girarda Sure, I'll hardcode it to `api` and remove the extra config value.